Storage/src/Connection/Rest.php

```diff
@@ -628,6 +629,53 @@ private function sendWithRetry($resource, $method, $args)
             $args
         );
 
+        $retryIdentifierHash = Uuid::uuid4()->toString();
+
+        $args['onRetryException'] = function (\Exception $e, $currentAttempt, &$arguments) use ($retryIdentifierHash) {
+            // Since we the the last attempt number here, so incrementing it
+            // to get the current attempt count
+            ++$currentAttempt;
+            $this->updateRetryInovationHeaders(
+                $arguments,
+                $retryIdentifierHash,
+                $currentAttempt
+            );
+        };
+
+        $args['onExecutionStart'] = function (&$arguments) use ($retryIdentifierHash) {
+            $this->updateRetryInovationHeaders(
+                $arguments,
+                $retryIdentifierHash
+            );
+        };
+
         return $this->send($resource, $method, $args);
     }
+
+    /**
+     * Updates the x-goog-api-client header value with UUID
```

Comment:
Instead of making this specifically for `x-goog-api-client` headers, we could make it generic. Let's accept this header name as an argument.

Reply:
Yeah, did this in this commit [af0e43c](https://github.com/googleapis/google-cloud-php/pull/5741/commits/af0e43c40fb3d06f1eb1649a437cf04305ab765a)

Thanks